binding/web/README.md

```diff
@@ -1,20 +1,297 @@
+import rhinoParams from "${PATH_TO_BASE64_RHINO_PARAMS}";
+
+const handle = await Rhino.fromBase64(
+  ${ACCESS_KEY},
+  { label: "rhino_model", base64: rhinoContext },
+  rhinoParams,
+  options // optional options
+)
+```
+
+### Process Audio Frames in Main Thread
+
+```typescript
+function getAudioData(): Int16Array {
+... // function to get audio data
+  return new Int16Array();
+}
+for (; ;) {
+  const inference = await handle.process(getAudioData());
+  if (inference.isFinalized) {
+    if (inference.isUnderstood) {
+      console.log(inference.intent)
+      console.log(inference.slots)
+      // Insert inference event callback here
+    }
+  }
+  // break on some condition
+}
+```
+
+### Initialize in Worker Thread
+
+Create a `inferenceDetectionCallback` function to get the streaming results from the worker:
+
+```typescript
+function inferenceDetectionCallback(inference) {
+  if (inference.isUnderstood) {
+    console.log(inference.intent)
+    console.log(inference.slots)
+  }
+}
+```
+
+Add to the `options` object an `processErrorCallback` function if you would like to catch errors:
+
+```typescript
+function processErrorCallback(error: string) {
+...
+}
+options.processErrorCallback = processErrorCallback;
+```
+
+Use `rhinoWorker` to initialize from public directory:
+
+```typescript
+const handle = await RhinoWorker.fromPublicDirectory(
+  ${ACCESS_KEY},
+  { label: "rhino_model", rhnPath: ${CONTEXT_RELATIVE_PATH} },
+  inferenceDetectionCallback,
+  ${MODEL_RELATIVE_PATH},
+  options // optional options
+);
+```
+
+or initialize using a base64 string:
+
+```typescript
+import rhinoContext from "${PATH_TO_BASE64_RHINO_CONTEXT}";
+import rhinoParams from "${PATH_TO_BASE64_RHINO_PARAMS}";
+
+const handle = await Rhino.fromBase64(
+  ${ACCESS_KEY},
+  { label: "rhino_model", base64: rhinoContext },
+  inferenceDetectionCallback,
+  rhinoParams,
+  options // optional options
+)
+```
+
+### Process Audio Frames in Worker Thread
+
+In a worker thread, the `process` function will send the input frames to the worker.
+The result is received from `inferenceDetectionCallback` as mentioned above.
+
+```typescript
+function getAudioData(): Int16Array {
+... // function to get audio data
+  return new Int16Array();
+}
+for (; ;) {
+  handle.process(getAudioData());
+  // break on some condition
+}
+```
+
+### Clean Up
+
+Clean up used resources by `Rhino` or `RhinoWorker`:
+
+```typescript
+await handle.release();
+```
+
+### Terminate
+
+Terminate `RhinoWorker` instance:
+
+```typescript
+await handle.terminate();
+```
+
+## Contexts
+
+Create custom contexts using the [Picovoice Console](https://console.picovoice.ai/).
+Train the Rhino context model for the target platform WebAssembly (WASM).
+Inside the downloaded `.zip` file, there are two files:
+
+- `.rhn` file which is the context model file in binary format
+- `_b64.txt` file which contains the same binary model encoded with Base64
```

Comment:
binary or .js (base64)

Reply:
we are though we just need the .rhn file now. probably not gonna separate wasm from others.